### PR TITLE
Sprint 2 favorite screen implementation

### PR DESCRIPTION
**Pull request by punAnish**
merge conflic on favorite screen implementation

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -100,9 +100,9 @@ dependencies {
     // Core and UI
     implementation("androidx.core:core-ktx:1.15.0")
     implementation("androidx.appcompat:appcompat:1.7.0")
-    implementation("com.google.android.material:material:1.12.0")
```
Comment (kiranpok) [inline app/build.gradle.kts:103]: typo

File: app/src/main/java/com/example/weathermate/data/WeatherDao.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ interface WeatherDao {
 
     // Get a specific city from the 'favorites' table
     @Query(value = "SELECT * FROM favorites WHERE city = :city")
-    suspend fun getFavoriteCityById(city: String): FavoriteCity
+    suspend fun getFavoriteCityById(city: String): FavoriteCity?
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/data/WeatherDao.kt:20]: for null

File: app/src/main/java/com/example/weathermate/data/WeatherDatabase.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@ import androidx.room.RoomDatabase
 import com.example.weathermate.model.FavoriteCity
 
 // Sets up the Room database for the app
-@Database(entities = [FavoriteCity::class], version = 1, exportSchema = false)
+@Database(entities = [FavoriteCity::class], version = 2, exportSchema = false)
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/data/WeatherDatabase.kt:8]: newest version

File: app/src/main/java/com/example/weathermate/di/Appmodule.kt [hunk 4]
```diff
@@ -40,11 +40,14 @@ class Appmodule {
         return Room.databaseBuilder(
             context,
             WeatherDatabase::class.java,
-            "weather_database")
+            "weather_database"
+        )
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/di/Appmodule.kt:44]: noth

File: app/src/main/java/com/example/weathermate/model/FavoriteCity.kt [hunk 5]
```diff
@@ -3,17 +3,21 @@ package com.example.weathermate.model
...
-    @Nonnull
     @PrimaryKey
     @ColumnInfo(name = "city")
     val city: String,
 
     @ColumnInfo(name = "country")
     val country: String,
+
+    @ColumnInfo(name = "temperature")
+    var temperature: Double? = null,
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/model/FavoriteCity.kt:19]: this is correct

File: app/src/main/java/com/example/weathermate/model/FavoriteCity.kt [hunk 6]
```diff
@@ -3,17 +3,21 @@ package com.example.weathermate.model
...
+
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/model/FavoriteCity.kt:20]: checked together

File: app/src/main/java/com/example/weathermate/model/FavoriteCity.kt [hunk 7]
```diff
@@ -3,17 +3,21 @@ package com.example.weathermate.model
...
+    @ColumnInfo(name = "weatherCondition")
+    var weatherCondition: String? = null
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/model/FavoriteCity.kt:22]: correct

File: app/src/main/java/com/example/weathermate/model/FavoriteCity.kt [hunk 8]
```diff
@@ -3,17 +3,21 @@ package com.example.weathermate.model
...
     @ColumnInfo(name = "city")
     val city: String,
 
     @ColumnInfo(name = "country")
     val country: String,
+
+    @ColumnInfo(name = "temperature")
+    var temperature: Double? = null,
+
+    @ColumnInfo(name = "weatherCondition")
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/model/FavoriteCity.kt:21]: correct

File: app/src/main/java/com/example/weathermate/repository/WeatherDbRepository.kt [hunk 9]
```diff
@@ -1,28 +1,36 @@
 package com.example.weathermate.repository
+import android.util.Log
 import com.example.weathermate.data.WeatherDao
 import com.example.weathermate.model.FavoriteCity
+import com.example.weathermate.model.Weather
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/repository/WeatherDbRepository.kt:5]: newest

**Review by kiranpok — APPROVED** [review]

File: app/build.gradle.kts [hunk 10]
```diff
@@ -100,9 +100,9 @@ dependencies {
...
+     implementation("com.google.android.material:material:1.12.0")
     implementation("androidx.compose.ui:ui:$composeVersion")
-    implementation("androidx.compose.material:material:$composeVersion")
```
Comment (Mamita123) [inline app/build.gradle.kts:105]: Need both.

File: app/src/main/java/com/example/weathermate/model/FavoriteCity.kt [hunk 11]
```diff
@@ -3,17 +3,21 @@ package com.example.weathermate.model
 import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
-import javax.annotation.Nonnull
 
 // Data class for FavoriteCity
 @Entity(tableName = "favorites")
 data class FavoriteCity(
 
-    @Nonnull
```
Comment (Mamita123) [inline app/src/main/java/com/example/weathermate/model/FavoriteCity.kt:12]: Remove this.

File: app/src/main/java/com/example/weathermate/repository/WeatherDbRepository.kt [hunk 12]
```diff
@@ -1,28 +1,36 @@
...
+import com.example.weathermate.network.WeatherApi
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 // Repository class that interacts with the database (between the ViewModel and the DAO)
-class WeatherDbRepository @Inject constructor(private val weatherDao: WeatherDao){
+class WeatherDbRepository @Inject constructor(private val weatherDao: WeatherDao, private val apiService: WeatherApi) {
```
Comment (Mamita123) [inline app/src/main/java/com/example/weathermate/repository/WeatherDbRepository.kt:13]: Add this.

**Review by Mamita123 — APPROVED** [review]
everything is ok.

File: app/src/main/java/com/example/weathermate/repository/WeatherDbRepository.kt [hunk 13]
```diff
@@ -32,8 +40,39 @@ class WeatherDbRepository @Inject constructor(private val weatherDao: WeatherDao
     }
 
     // Fetches a favorite city by its name from the database
-    suspend fun getFavoriteCityById(city: String): FavoriteCity {
-        return weatherDao.getFavoriteCityById(city)
+    suspend fun getFavoriteCityById(city: String): FavoriteCity? {
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/repository/WeatherDbRepository.kt:43]: this is correct

File: app/src/main/java/com/example/weathermate/screens/favorites/FavoriteCityScreen.kt [hunk 14]
```diff
@@ -109,38 +113,110 @@ fun FavoriteCityScreen(
     }
 }
 
-// function to display the favorite city row
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/screens/favorites/FavoriteCityScreen.kt:112]: ok

File: app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt [hunk 15]
```diff
@@ -16,8 +17,6 @@ import androidx.compose.material.Text
 import androidx.compose.material.TopAppBar
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
-import androidx.compose.runtime.collectAsState
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt:19]: needed

File: app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt [hunk 16]
```diff
@@ -16,8 +17,6 @@ import androidx.compose.material.Text
...
-import androidx.compose.runtime.getValue
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt:20]: needed

File: app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt [hunk 17]
```diff
@@ -38,12 +37,13 @@ import com.example.weathermate.navigation.WeatherScreens
 import android.widget.Toast
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.filled.Favorite
-import androidx.compose.material.icons.filled.FavoriteBorder
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt:41]: needed

File: app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt [hunk 18]
```diff
@@ -131,33 +132,53 @@ fun WeatherMateAppBar(
...
+                            if (dataList.size < 2) {
+                                // Debugging: Log invalid title format
+                                Log.e("WeatherMateAppBar", "Invalid title format: $title")
+                                Toast.makeText(context, "Invalid city format", Toast.LENGTH_SHORT).show()
+                                return@clickable
+                            }
                             val cityName = dataList[0]
                             val countryName = dataList[1]
 
-                            // Check if the city is already in the favorite list
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/widgets/WeatherMateAppBar.kt:137]: comment

File: app/src/main/java/com/example/weathermate/screens/favorites/FavoriteCityViewModel.kt [hunk 19]
```diff
@@ -13,60 +13,98 @@ import kotlinx.coroutines.flow.distinctUntilChanged
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-// ViewModel class that interacts with the repository to manage the favorite cities
```
Comment (kiranpok) [inline app/src/main/java/com/example/weathermate/screens/favorites/FavoriteCityViewModel.kt:16]: comment

**Review by kiranpok — APPROVED** [review]

**Comment by kiranpok** [comment]
we have reviewed the changes, you can merge it.

**Comment by Mamita123** [comment]
Review done.

**Pull request merged**